pkg/application/application.go

```diff
@@ -10,12 +10,28 @@ import (
 	"github.com/redhat-developer/odo/pkg/config"
 	"github.com/redhat-developer/odo/pkg/occlient"
 	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/redhat-developer/odo/pkg/util"
 )
 
-// getDefaultAppName returns application name to be used as a default name in case the user doesn't provide a name.
-// In future this function should generate name with unique suffix (app-xy1h), because there might be multiple applications.
-func getDefaultAppName() string {
-	return "app"
+// GetDefaultAppName returns randomly generated application name with unique configurable prefix suffixed by a randomly generated string which canbe used as a default name in case the user doesn't provide a name.
+func GetDefaultAppName(existingAppNames []string) (string, error) {
```

Comment:
Instead of []string you can use whatever variable `apps` is (I believe it's a struct?)

Reply:
Done